### DISCUSSION TITLE
Can only download Video, no Audio

### DISCUSSION THREAD

**Discussion by Pixoticelixer** [aclap-dev/video-downloadhelper#1208] [Help and Q&A]
### Your question

Hello,

I used to be able to freely and efficiently download videos from motogp.com. Now with this extension it only downloads Video files with no Audio stream. Also, when i try to download Audio separately it fails and gives me an error.
Although it says its going to download both together, it never does. Just Video.

Any help is appreciated. Thanks.

![Screenshot 2024-09-03 174048](https://github.com/user-attachments/assets/49cc86d0-a698-416c-84c2-2c8864eebd67)
![Screenshot 2024-09-03 173852](https://github.com/user-attachments/assets/8b06b1ff-1637-4ada-9467-0e34b0b96592)

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"e0c2bdd9b85642c7"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "E:\\uTorrent\\!done\\yt",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive"
        }
    },
    "view_options": {
        "all_tabs": true,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_wide_ui": true,
    "use_legacy_ui": false,
    "use_sidebar": true,
    "open_count_store": 2,
    "successfull_dl": 4,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_4182605943162503"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "E:\\uTorrent\\!done\\yt\\Yamahas new Technical Director targets 2026 for return to the to-01.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Yamahas new Technical Director targets 2026 for return to the to-01.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "E:\\uTorrent\\!done\\yt"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.motogp.com/en/videos/2024/09/03/yamahas-new-technical-director-targets-2026-for-return-to-the-top/506858?playlistId=216686"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1725380526450
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_5981214025846674"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "E:\\uTorrent\\!done\\yt\\Gresinis Masini Almost everybody started to cry.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Gresinis Masini Almost everybody started to cry.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "E:\\uTorrent\\!done\\yt"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.motogp.com/en/videos/2024/09/01/gresinis-masini-almost-everybody-started-to-cry/507263?playlistId=216686"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1725380855921
                        }
                    }
                ]
            }
        ]
    },
    "record_download_history": true
}

**Comment by GrampaWildWilly** [comment]
#625.  Others have been there done that.

**Reply by Pixoticelixer** [reply]
Doesnt seemed to be solved at all. 

More of a suggestion to study ffmpeg. Isnt the whole point of VDH to help users that dont have a degree in software development?

**Reply by GrampaWildWilly** [reply]
You're right.  VDH does not work properly on that site.  That leaves you with a simple choice.  You can wait for some unknown length of time for @paulrouget Paul to fix VDH.  Or you can get the content now using the approach I outlined in #625.  Or you can find another tool that does work on that site, if such a tool exists.  You don't need a degree in software engineering to use ffmpeg.  You just need to decide that you can do it & then you will do it.  It's a question of attitude, not ability.  If you don't want to bother, that's also your choice.  I'm not trying to make you or anybody learn anything.  I'm just offering you the tools.  It's up to you whether you learn to use them.

**Reply by Pixoticelixer** [reply]
You're right, it doesn't require a software degree but it would surely help. Ffmpeg is extremely complicated for people who are not well versed in cmd commands. 

I guess I'm looking for a quick solution to this issue. I have tried to follow your detailed tips in #625, but to no avail.

I will try doing more research into ffmpeg, hopefully it is the solution.

**Reply by GrampaWildWilly** [reply]
ffmpeg looks like this:

ffmpeg &lt;parameters&gt; -i "input thing" &lt;some more parameters&gt; "output file"

The "input thing" is usually the URL of a manifest when  you're doing a download, but it can be a file on your system when you're doing some other manipulation.  And there are more possibilities, most of which I don't know because ffmpeg is so atrociously badly documented.  The parameters took me a while, lots of trial & error, to come up with.  When I post a log, it always contains the exact text of the command I execute.  Don't think about those.  Just use the ones I show.

**Reply by Pixoticelixer** [reply]
I will try this, thanks.

If I am successful I will be sure to document it here.

**Reply by GrampaWildWilly** [reply]
Also if you're unsuccessful.  A mistake is a learning opportunity.
